src/main/java/de/hhu/propra/sharingplatform/service/ApiService.java

```diff
@@ -0,0 +1,105 @@
+package de.hhu.propra.sharingplatform.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import de.hhu.propra.sharingplatform.dao.PaymentRepo;
+import de.hhu.propra.sharingplatform.dto.ProPay;
+import de.hhu.propra.sharingplatform.dto.ProPayReservation;
+import de.hhu.propra.sharingplatform.model.Payment;
+import de.hhu.propra.sharingplatform.model.User;
+import lombok.Data;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.io.BufferedReader;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+@Data
+@Service
+public class ApiService {
+
+    private PaymentRepo paymentRepo;
+    String host = "localhost";
+
+    @Autowired
+    public ApiService(PaymentRepo paymentRepo) {
+        this.paymentRepo = paymentRepo;
+    }
+
+    String fetchJson(String userName) {
+        String url = "http://" + host + ":8888/account/" + userName;
+        RestTemplate jsonResponse = new RestTemplate();
+
+        return jsonResponse.getForObject(url, String.class);
+    }
+
+    public ProPay mapJson(String userName) {
+        String jsonResponse = fetchJson(userName);
+        ObjectMapper mapper = new ObjectMapper();
+
+        try {
+            return mapper.readValue(jsonResponse, ProPay.class);
+        } catch (IOException ioException) {
+            ioException.printStackTrace();
+            return null;
+        }
+    }
+
+    public void enforcePayment(Payment payment, double totalPrice) {
+        long id = reservateMoney(payment.getProPayIdSender(), payment.getProPayIdRecipient(),
+            payment.getBail());
+        payment.setBailProPayId(id);
+        id = reservateMoney(payment.getProPayIdSender(), payment.getProPayIdRecipient(),
+            totalPrice);
+        payment.setAmountProPayId(id);
+        paymentRepo.save(payment);
+    }
+
+    private long reservateMoney(String proPayIdSender, String proPayIdRecipient, double amount) {
+        try {
+            URL url =
+                new URL("http://localhost:8888/reservation/reserve/" + proPayIdSender
```

Comment:
use the defined variable instead of hardcoding localhost

Reply:
Sure will be done. Forgot to set it properly.